### ISSUE TITLE
Nextcloud destop 3.3.x AppImage doesn't start if virtual file support is enabled

### ISSUE THREAD

**Issue by jacek-kubek** [nextcloud/desktop#3781]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Expected behaviour

Nextcloud desktop 3.3.x AppImage starts like Nextcloud desktop 3.2.x does if virtual file support is enabled.

### Actual behaviour

After enabling virtual file support in Nextcloud desktop 3.2.x AppImage, it  works like expected.
After upgrading to Nextcloud desktop 3.3.x AppImage, it tries to start, but fails with error.

The same behaviour is reported [here](https://www.mail-archive.com/debian-bugs-dist@lists.debian.org/msg1815724.html).

### Steps to reproduce

1. Download [Nextcloud-3.2.4-x86_64.AppImage](https://download.nextcloud.com/desktop/releases/Linux/Nextcloud-3.2.4-x86_64.AppImage).
2. Make it executable and run it (or use AppImageLauncher and choose ```Run once```).
3. Enable virtual file support.
4. Close it.
5. Download [Nextcloud-3.3.3-x86_64.AppImage](https://download.nextcloud.com/desktop/releases/Linux/Nextcloud-3.3.3-x86_64.AppImage).
6. Make it executable and run it (or use AppImageLauncher and choose ```Run once```).

### Client configuration

Client version:
3.3.3

Operating system:
Linux Mint Debian Edition 4 (Debbie)

OS language:
en_US

Qt version used by client package (Linux only, see also Settings dialog):
5

Client package (From Nextcloud or distro) (Linux only):
[Nextcloud-3.3.3-x86_64.AppImage](https://download.nextcloud.com/desktop/releases/Linux/Nextcloud-3.3.3-x86_64.AppImage)

Installation path of client:
$HOME/Applications/

### Server configuration
No matter.

### Logs

1. Client logfile:
No log file, just error message during start.
```
$ ./Nextcloud-3.3.3-x86_64.AppImage 
qt5ct: using qt5ct plugin
2021-09-10 17:35:08:653 [ critical plugins ]:	Could not load plugin: not existant or bad metadata "nextcloudsync_vfs_suffix"
2021-09-10 17:35:08:653 [ fatal default ]:	Could not load plugin
```

2. Web server error log:
No matter.

3. Server logfile: nextcloud log (data/nextcloud.log):
No matter.

**Comment by mgallien** [comment]
We have fixed that problem in the latest release.
Can you test with 3.3.3 version ?

**Comment by nursoda** [comment]
I'm on arch, installed regular `nextcloud-client` (3.3.3) and added some accounts. Then I wanted to enable virtual files. I first added `showExperimentalOptions=true` to `nextcloud.cfg` and then tried to `Activate virtual files` in one of the clouds. App reproducibly crashes doing that, and when restarted, it's in the state as before (virtual files disabled). So, this might be the same issue.

**Comment by jacek-kubek** [comment]
@mgallien
There is bug in all versions of client in the 3.3.x 'line', and there is no change in 3.3.3 (as I mentioned in my initial post).
The client can not load (find?) `nextcloudsync_vfs` plugin:
```
$ ./Nextcloud-3.3.3-x86_64.AppImage 
qt5ct: using qt5ct plugin
2021-09-13 20:00:09:107 [ critical plugins ]:	Could not load plugin: not existant or bad metadata "nextcloudsync_vfs_suffix"
2021-09-13 20:00:09:108 [ fatal default ]:	Could not load plugin
```
As far as I observed, the 3.2.x client doesn't need this plugin:
```
$ ./Nextcloud-3.2.4-x86_64.AppImage 
qt5ct: using qt5ct plugin
```

@nursoda
In my opinion it is the same issue.
Just give a try [the last version of client in the 3.2.x 'line'](https://download.nextcloud.com/desktop/releases/Linux/Nextcloud-3.2.4-x86_64.AppImage). It works for me without this issue and I have to stay with it.

**Comment by FlexW** [comment]
@mgallien it was only fixed in the build-appimage script in this repository. It was not fixed in brander. That's why the error is still in 3.3.3.

@nursoda Ou build/install process via make/ninja is currently broken and packagers need to copy the vfs plugins into the correct location by themself. See also https://github.com/nextcloud/desktop/blob/be8f60fa8ec863771afa84bdc1bf6cb61d6d9896/admin/linux/build-appimage.sh#L53

**Comment by njakes** [comment]
On a fresh install I am also having issues with this.  When adding showExperimentalOptions=true to the nextcloud.cfg it makes the option available.  However, as soon as Enable virtual file support is selected the entire application crashes with a Segmentation fault (core dumped) output in terminal. Is there a way to manually edit 0\Folders\1\virtualFilesMode=no in nextcloud.cfg?  
This is also running the latest appimage from the releases page here.
If I clear out the logs and try again the last line of the log states:
[ warning default ]:  QObject::connect(QDialog, OCC::AccountSettings): invalid null parameter.

I hope this helps in some way.

**Comment by nursoda** [comment]
[Fixed](https://bugs.archlinux.org/task/72133) in Arch package [nextcloud-client](https://archlinux.org/packages/community/x86_64/nextcloud-client/) 3.3.3-2.

**Comment by mgallien** [comment]
Next release 3.3.4 should have a working virtual files support in the Appimage build.

Once you have it, can you test and close the issue if the bug is fixed ?

**Comment by jacek-kubek** [comment]
@mgallien
I am waiting for 3.3.4 release.
The [3.3.3](https://github.com/nextcloud/desktop/releases/tag/v3.3.3) one is still the `Latest release`.

**Comment by jacek-kubek** [comment]
I can confirm the issue with `nextcloudsync_vfs` plugins is resolved in `Nextcloud-3.3.4-x86_64.AppImage` release.

**Comment by antonio-rojas** [comment]
This is still not fixed in 1.3.4 when building from source. The plugins are installed in some PLUGINDIR path which is nowhere defined

https://github.com/nextcloud/desktop/blob/master/src/libsync/vfs/suffix/CMakeLists.txt#L29

**Comment by mgallien** [comment]
> This is still not fixed in 1.3.4 when building from source. The plugins are installed in some PLUGINDIR path which is nowhere defined
> 
> https://github.com/nextcloud/desktop/blob/master/src/libsync/vfs/suffix/CMakeLists.txt#L29

yes and sorry for that
we wanted to fix it but had no time finding the proper combination that would work for all platforms
